core/src/main/resources/org/teavm/backend/javascript/runtime.js

```diff
@@ -15,15 +15,10 @@
  */
 "use strict";
 var $rt_global = this;
-var $rt_lastObjectId = 1;
 function $rt_nextId() {
-    var current = $rt_lastObjectId;
-    var next = (current + 1) | 0;
-    if (next === 0) {
-        next = (next + 1) | 0;
-    }
-    $rt_lastObjectId = next;
-    return current;
+    var min = -0x80000000;
+    var max = 0x7fffffff;
+    return Math.floor(Math.random() * (max - min + 1)) + min;
```

Comment:
I don't like this. It should affect performance, right? Perhaps there should be better approach (i.e. rotate by fixed ratio and xor by fixed value).